### PR TITLE
Use google.cloud.logging for logging on App Engine.

### PR DESCRIPTION
**Pull request by oliverchang**
- Install google.cloud.logging handler to root logger. This is necessary for severity logging to work.
- Also attach trace information to each log to allow correlating logs with requests.

### CODE REVIEW

File: src/python/metrics/logs.py [hunk 1]
```diff
@@ -252,14 +252,28 @@ def uncaught_exception_handler(exception_type, exception_value,
   sys.__excepthook__(exception_type, exception_value, exception_traceback)
 
 
+def configure_appengine():
+  """Configure logging for App Engine."""
+  logging.getLogger().setLevel(logging.INFO)
+
+  if os.getenv('LOCAL_DEVELOPMENT') or sys.version_info.major == 2:
```
Comment (inferno-chromium) [inline src/python/metrics/logs.py]: split into two checks to make it easy to not to remove os.getenv('LOCAL_DEVELOPMENT') check.

File: src/python/metrics/logs.py [hunk 2]
```diff
@@ -317,12 +331,37 @@ def get_source_location():
   return 'Unknown', '-1', 'Unknown'
 
 
+def _add_appengine_trace(extras):
+  """Add App Engine tracing information."""
+  if sys.version_info.major != 3 or not _is_running_on_app_engine():
+    # TODO(ochang): Remove Python 3 check once all migrated to Python 3.
```
Comment (inferno-chromium) [inline src/python/metrics/logs.py]: Does it break anything on python 2, if not can simplify this check and one above to remove sys.version_info.major != 3

File: src/python/metrics/logs.py [hunk 3]
```diff
@@ -317,12 +331,37 @@ def get_source_location():
...
+          project_id=project_id, trace_id=trace_id)
+
+
 def emit(level, message, exc_info=None, **extras):
   """Log in JSON."""
   logger = get_logger()
   if not logger:
     return
 
+  if exc_info == (None, None, None):
```
Comment (inferno-chromium) [inline src/python/metrics/logs.py]: Why this, can you add a comment, is this needed for appengine (not bot)?

File: src/python/metrics/logs.py (hunk 1 above)
Comment (inferno-chromium) [reply src/python/metrics/logs.py]: more of a nit, can skip.

File: src/python/metrics/logs.py (hunk 2 above)
Comment (oliverchang) [reply src/python/metrics/logs.py]: Want to avoid any unexpected breakages there. Will be removed very soon anyway.

**Comment by oliverchang** [comment]
Btw I think with this logging library we can get rid of fluentd completely. Will file a bug for that.

**Pull request merged**